src/tcp.toit

```diff
@@ -100,7 +100,10 @@ class ReconnectingTransport_ extends TcpTransport:
     port_ = port
     open_ = net-open
     super.from-subclass_
-    reconnect
+    try:
+      reconnect
+    finally: | is-exception _ |
+      if is-exception: close
```

Comment:
Not sure if this is better, but at least this way we can avoid relying on `is-exception` in more places.

```suggestion
    success := false
    try:
      reconnect
      success = true
    finally:
      if not success: close
```

Alternatively you can also consider using `catch` because you only care about the exceptional case. You want to continue the unwinding so it isn't going to be super pretty.

```suggestion
    catch --unwind=(: close; true): 
      reconnect
```